run_tests.sh

```diff
@@ -1,6 +1,9 @@
 #!/bin/bash -ex
 
 export PIPENV_IGNORE_VIRTUALENVS=1
+set +x  # Keep the API key out of execution logs
+export DEPSDEV_API_KEY=$(kubectl get secret secrets -o jsonpath='{.data.deps\.dev}' | base64 --decode)
```

Comment:
Does this require a `gcloud container clusters get-credentials ...` call before? The secret is tied to a GKE cluster right.

Reply:
> Does this require a `gcloud container clusters get-credentials ...` call before? The secret is tied to a GKE cluster right.

All I can say definitively is it would appear to be working correctly in Cloud Build (I imagine because it's running on GCP, there's ambient access to the necessary credentials)

Reply:
Interesting, I was curious because there can be multiple GKE clusters in a single GCP project, so it's not clear how this is able to determine the right one..

Reply:
I wonder if this is just failing (i.e. the key is getting assigned an empty string) and  tests pass because you added `unittest.skipIf`. 

Doing a local  test and adding this to `run_tests.sh`:

```bash
export FOO=$(false)
```

Things seem to proceed despite the `-e`. `FOO` just gets set as empty.

Reply:
> I wonder if this is just failing (i.e. the key is getting assigned an empty string) and tests pass because you added `unittest.skipIf`.

I wouldn't have expected `test_maven_deps_dev` to then pass if `DEPSDEV_API_KEY` was set to an unusable value?

Reply:
A failed kubectl get secret command seems to give me an empty string to stdout (with error logs in stderr), so it's possible this is is being passed through as an empty string?

Reply:
> A failed kubectl get secret command seems to give me an empty string to stdout (with error logs in stderr), so it's possible this is is being passed through as an empty string?

If that is the case, why does `test_maven_deps_dev` pass?

Reply:
Is there a way we can list out the secrets (without revealing them) on GCB to verify this is definitely the right one?

Reply:
Switching over to Cloud Secrets in b06550f6db9bdb5200125117f794f130b098853a